### PR TITLE
Variance aware neural network training (also outputs probability distribution over spike rates)

### PR DESCRIPTION
**Pull request by thomasZen**
This lead to worse performance then the Poisson loss without considering variances. I opened this PR only to describe what I did, not for merging.

- Estimate variance over test repeats for each neuron
- Discretize learning rate to in 0.1 steps between 0.0 and 15.0
- Train network with cross entropy loss and normal probability distribution with mean=measured firing rate, variance=estimated variance of neuron

During decoding:
- Argmax: Use the highest probability firing rate as the firing rate prediction
- Soft: Multiply the firing rate bins with their probability to get a firing rate prediction

Results:
```
       Test metric             Train             Validation             Test                                                                                 
──────────────────────────────────────────────────────────────────────────────────────────────────
       correlation          0.1440441757440567       0.18190255761146545      0.32424962520599365                                                                              
    correlation_soft        0.22398288547992706      0.27573615312576294      0.4322403371334076                                                                               
          loss              3.3493404388427734        3.339628219604492        3.329080581665039                                                                               
```

### CODE REVIEW

**Comment by thomasZen** [comment]
Closing as it's outdated.
